### PR TITLE
finish up Type writers

### PR DESCRIPTION
**Pull request by dmitrykozinets**

### CODE REVIEW

File: morphir/core/test/src/org/finos/morphir/ir/WritersSpec.scala [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        ) +
+          suite("ExtensibleRecordTypeWriter")(
+            test("Should support writing Morphir IR Type ExtensibleRecord")(
+              expectAllEqual(
+                ExtensibleRecord[Int](
+                  1,
+                  Name("ExtRec"),
+                  List(Field[Int]("key1", Variable[Int](2, Name("x"))), Field[Int]("key2", Variable[Int](2, Name("y"))))
+                )
+                  -> """["ExtensibleRecord",1,["ext","rec"],[{"key1":["Variable",2,["x"]]},{"key2":["Variable",2,["y"]]}]]"""
```
[the remark this answers is not in the document]
Comment (DamianReeves) [reply morphir/core/test/src/org/finos/morphir/ir/WritersSpec.scala:89]: I think this is off see: 
https://github.com/finos/morphir-elm/blob/792dca4158a779617bfda40051a0810dc02c38e8/src/Morphir/IR/Type/Codec.elm#L149-L154

File: morphir/core/src/org/finos/morphir/ir/writers.scala [hunk 2]
```diff
@@ -46,9 +85,27 @@ trait IRTypeWriters extends NamingWriters { self: Annotator =>
...
+      case r: T.Type.Record[A]           => RecordTypeWriter.write0(out, r)
+      case r: T.Type.Reference[A]        => ReferenceTypeWriter.write0(out, r)
+      case r: T.Type.Tuple[A]            => TupleTypeWriter.write0(out, r)
+      case r: T.Type.Unit[A]             => UnitTypeWriter.write0(out, r)
+      case r: T.Type.Variable[A]         => VariableTypeWriter.write0(out, r)
+    }
+  }
 
-  implicit def FieldTypeWriter[A: Writer]: Writer[T.Field[A]] = ???
+  implicit def FieldTypeWriter[A: Writer]: Writer[T.Field[A]] = new Writer[T.Field[A]] {
```
Comment (DamianReeves) [inline morphir/core/src/org/finos/morphir/ir/writers.scala:100]: With a field we should get output that conforms to this:

https://github.com/finos/morphir-elm/blob/792dca4158a779617bfda40051a0810dc02c38e8/src/Morphir/IR/Type/Codec.elm#L149-L154

**Review by DamianReeves — CHANGES_REQUESTED** [review]

File: morphir/core/test/src/org/finos/morphir/ir/WritersSpec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmitrykozinets) [reply morphir/core/test/src/org/finos/morphir/ir/WritersSpec.scala]: fixed

File: morphir/core/src/org/finos/morphir/ir/writers.scala (hunk 2 above)
Comment (dmitrykozinets) [reply morphir/core/src/org/finos/morphir/ir/writers.scala:100]: fixed

**Review by DamianReeves — APPROVED** [review]
LGTM

**Pull request merged**